cpp/src/barretenberg/honk/sumcheck/sumcheck_output.hpp

```diff
@@ -8,12 +8,38 @@ namespace proof_system::honk::sumcheck {
  * @brief Contains the multi-linear evaluations of the polynomials at the challenge point 'u'.
  * These are computed by the prover and need to be checked using a multi-linear PCS like Gemini.
  */
-template <typename FF> struct SumcheckOutput {
+template <typename Flavor> struct SumcheckOutput {
+    using FF = typename Flavor::FF;
+    using PurportedEvaluations = typename Flavor::PurportedEvaluations;
     // u = (u_0, ..., u_{d-1})
     std::vector<FF> challenge_point;
     // Evaluations in `u` of the polynomials used in Sumcheck
-    std::array<FF, honk::StandardArithmetization::NUM_POLYNOMIALS> evaluations;
+    PurportedEvaluations purported_evaluations;
 
-    bool operator==(const SumcheckOutput& other) const = default;
+    SumcheckOutput()
+        : purported_evaluations(std::array<FF, Flavor::NUM_ALL_ENTITIES>()){};
+
+    SumcheckOutput(const std::vector<FF>& _challenge_point, const PurportedEvaluations& _purported_evaluations)
+        : challenge_point(_challenge_point)
+        , purported_evaluations(_purported_evaluations){};
+
+    SumcheckOutput& operator=(SumcheckOutput&& other)
+    {
+        challenge_point = other.challenge_point;
+        purported_evaluations = other.purported_evaluations;
+        return *this;
+    };
+
+    SumcheckOutput(const SumcheckOutput& other)
+        : challenge_point(other.challenge_point)
+        , purported_evaluations(other.purported_evaluations){};
+
+    bool operator==(const SumcheckOutput& other) const
+    {
+        bool result{ false };
+        result = challenge_point == other.challenge_point;
+        // result = purported_evaluations == other.purported_evaluations;
```

Comment:
Re-enable this!

Reply:
The issue is that there is no `operator==` for `PurportedEvaluations`. Rather than implementing, I just compare the underlying data.